.github/workflows/pull-request.yml

```diff
@@ -79,7 +79,28 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
     - name: Test with tox
-      run: tox
+      run: |
+        tox
+
+        # make sure python docs are up-to-date
+        if [[ "${{ matrix.python-version }}" == "3.8" ]]; then
```

Comment:
we only do this for 3.8 in the matrix because that version already receives special treatment in tox:
https://github.com/projectnessie/nessie/blob/42342a27ecf2edc65eee1c2d4536656de67e47b5/python/tox.ini#L20-L25

Reply:
Hm. Is it possible to add this check to tox.ini instead of a special case here?

Reply:
Or better: Why not just unconditionally add `docs` to the gh-actions for 3.8?

Reply:
i already tried to explain in the PR description why i did not want to put this into tox, but if we are fine with reasoning about git status in there then i will move it

Reply:
Yea, I think, that's okay.
If the generated docs are different from the checked-in docs, that's IMO a "bug".
I'd be even fine to move it into the `docs` part.

Reply:
> I'd be even fine to move it into the docs part.

then you will get an error everytime you run `tox -e docs` to update to the latest docs locally, so not a good idea imo

the "code" is simpler now but its only a valid up-to-date check in the following context:
- clean git repo state
- you run `docs` env just before

these conditions are mostly only guaranteed in CI, which is why i put the more verbose logic there.
its just a matter of different trade-offs, so i hope the current solution is fine now?

Reply:
Hm, yea - that's practically a bit odd.